### PR TITLE
Improve IpcParsedCommand domain message info.

### PR DESCRIPTION
**Pull request by SciresM**
Currently it's hard to use IpcParsedCommand for domain messages.

This fixes that.

### CODE REVIEW

File: nx/include/switch/kernel/ipc.h [hunk 1]
```diff
@@ -45,9 +45,24 @@ typedef enum {
...
     IpcCommandType_RequestWithContext = 6,
-    IpcCommandType_ControlWithContext = 7
+    IpcCommandType_ControlWithContext = 7,
 } IpcCommandType;
 
+typedef enum {
+    DomainMessageType_Invalid = 0,
+    DomainMessageType_SendMessage = 1,
+    DomainMessageType_Close = 2,
+} DomainMessageType;
```
Comment (yellows8) [inline nx/include/switch/kernel/ipc.h]: Ipc prefix missing + elsewhere.

Comment (SciresM) [reply nx/include/switch/kernel/ipc.h]: I was matching the existing style -- @plutooo called the domain header type "DomainMessageHeader", does that also need to change?

File: nx/include/switch/kernel/ipc.h [hunk 2]
```diff
@@ -618,8 +629,20 @@ static inline Result ipcParseForDomain(IpcParsedCommand* r) {
...
     r->Raw = (void*)(((uintptr_t) r->Raw) + sizeof(DomainMessageHeader));
 
+    r->IsDomainMessage = true;
+    r->MessageType = (DomainMessageType)(hdr->Type);
+    switch (r->MessageType) {
+        case DomainMessageType_SendMessage:
+        case DomainMessageType_Close:
+            break;
+        default:
+            return 0xF601;
```
Comment (yellows8) [inline nx/include/switch/kernel/ipc.h]: Does official sw return this?

File: nx/include/switch/kernel/ipc.h [hunk 3]
```diff
@@ -636,12 +636,12 @@ static inline Result ipcParseForDomain(IpcParsedCommand* r) {
         case DomainMessageType_Close:
             break;
         default:
-            return 0xF601;
+            return 0x1A60A;
```
Comment (yellows8) [inline nx/include/switch/kernel/ipc.h]: Change these to a new libnx error.

Comment (fincs) [reply nx/include/switch/kernel/ipc.h]: I think this is supposed to match official sw, not sure.

**Pull request merged**